BCStudentSoftwareDevTeam/celts
# PR for Warning label track hours issue 45(This issue is based off of branch: track_volunteer_hours_issue42)

This PR fixes issue #45 

Summary: 
 The issue asked that there be a way for an admin to tell that a participant that has checked into or rsvp to an event be flagged if they have not completed all prerequisites for the event. In order to achieve this, we queried the database and pushed on to a list of all the eventIDs that were set as prerequisites to the program the participant is attending. We then queried the database using the eventIDS and pushed onto a list all the users who attended any of the prerequisite events, using this list we then checked how many times a user name appeared and created a conditional that required the user to appear as many times as the length of the eventIDs list. If the user met the amount they would stay on the list, if the user did not they would be removed from the list. After deleting duplicates we then passed the list to the HTML and used jinja to flag any users who appeared on the track hours page who were not on the list of eligible participants. 

To Test:  
There is a test function in the test suite labeled `test_warning.py`. Running this should pass.
You can also visit the URL: `http://<YOURIP>/<PROGRAMID>/<EVENTID>/track_hours`. This URL will display a table with every user set to attend or currently at the event, if any of them have not met the requirements for the event you will see a red warning icon. Hovering over this will display a ToolTip that will tell you that the user has not completed the prerequisites.

Notes: This branch is forked off over the branch listed in the title. If any changes are made to the listed branch this will have to be updated.
